README.md

```diff
@@ -23,37 +28,31 @@ So, because root depends on both menu >=1.0.0 and intl >=5.0.0,
   version solving failed.
 ```
 
-The algorithm is generic and works for any type of dependency system
+This pubgrub crate provides a Rust implementation of PubGrub.
+It is generic and works for any type of dependency system
 as long as packages (P) and versions (V) implement
-the `Package` and `Version` traits.
-`Package` is strictly equivalent and automatically generated
-for any type that implement `Clone + Eq + Hash + Debug + Display`.
-`Version` simply states that versions are ordered,
-that there should be
-a minimal `lowest` version (like 0.0.0 in semantic versions),
-and that for any version, it is possible to compute
-what the next version closest to this one is (`bump`).
-For semantic versions, `bump` corresponds to an increment of the patch number.
-
+the provided `Package` and `Version` traits.
 
-## API
 
-```rust
-let solution = resolve(&dependency_provider, package, version)?;
-```
+## Using the pubgrub crate
 
-Where `dependency_provider` supplies the list of available packages and versions,
-as well as the dependencies of every available package
-by implementing the `DependencyProvider` trait.
-The call to `resolve` for a given package at a given version
-will compute the set of packages and versions needed
-to satisfy the dependencies of that package and version pair.
-If there is no solution, the reason will be provided as clear as possible.
+A [guide][guide] with both high-level explanations and
+in-depth algorithm details is available online.
+The [API documentation is available on docs.rs][docs].
+A WIP version of the [API docs for the dev branch][docs-dev] is also
+accessible for convenience.
 
 
 ## Contributing
 
-Discussion and development happens here on github and on our [Zulip stream](https://rust-lang.zulipchat.com/#narrow/stream/260232-t-cargo.2FPubGrub). Please join in!
+Discussion and development happens here on github and on our
+[Zulip stream](https://rust-lang.zulipchat.com/#narrow/stream/260232-t-cargo.2FPubGrub).
+Please join in!
```

Comment:
```suggestion
Discussion and development happens here on Github and on our
[Zulip stream](https://rust-lang.zulipchat.com/#narrow/stream/260232-t-cargo.2FPubGrub).
Please join in!

```

Reply:
`Github` -> `GitHub`

Reply:
Huh, I never noticed they spell it this way! Alright